### PR TITLE
Fix stress test: check if storage shutdown before we operate MergeTreeDeduplicationLog

### PR DESCRIPTION
**Pull request by hanfei1991**
fix ci #47641
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit dccbe875d247818a17e999ceab5e062537169f80 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52623/dccbe875d247818a17e999ceab5e062537169f80/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp [hunk 1]
```diff
@@ -231,6 +233,12 @@ std::pair<MergeTreePartInfo, bool> MergeTreeDeduplicationLog::addPart(const std:
         return std::make_pair(info, false);
     }
 
+    if (stopped)
+    {
+        LOG_ERROR(&Poco::Logger::get("MergeTreeDeduplicationLog"), "Storage has been shutdown when we add this part.");
+        return {};
```
Comment (tavplubix) [inline src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp:239]: Can this block be committed after this? Should we throw an exception instead?

Comment (tavplubix) [reply src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp:239]: `UNFINISHED` or `ABORTED`

File: src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp [hunk 2]
```diff
@@ -10,9 +10,17 @@
 #include <Disks/WriteMode.h>
 #include <Disks/IDisk.h>
 
+#include "Common/Exception.h"
```
Comment (hanfei1991) [inline src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp:13]:
```suggestion
#include <Common/Exception.h>
```

File: src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp [hunk 3]
```diff
@@ -231,6 +239,11 @@ std::pair<MergeTreePartInfo, bool> MergeTreeDeduplicationLog::addPart(const std:
         return std::make_pair(info, false);
     }
 
+    if (stopped)
+    {
+        throw Exception(ErrorCodes::ABORTED, "Storage has been shutdown when we drop this part.");
```
Comment (hanfei1991) [inline src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp:244]:
```suggestion
        throw Exception(ErrorCodes::ABORTED, "Storage has been shutdown when we add this part.");
```

File: src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp [hunk 4]
```diff
@@ -10,9 +10,17 @@
 #include <Disks/WriteMode.h>
 #include <Disks/IDisk.h>
 
+#include <Common/Exception.h>
+#include <Common/logger_useful.h>
```
Comment (hanfei1991) [inline src/Storages/MergeTree/MergeTreeDeduplicationLog.cpp:14]:
```suggestion
```

**Review by tavplubix — APPROVED** [review]

**Pull request merged**